src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/routeRegistry.js

```diff
@@ -23,6 +24,15 @@ class RouteRegistry {
                 ...routeConfig,
                 children: [],
                 parent: undefined,
+                get availableAttributes() {
```

Comment:
Have you also tested if these getters are really cached? Will test it later as well, just to be sure.

Reply:
I know I am a little bit late to recognize this, but it seems I didn't think about the impact of this enough... Especially when I have a look at the tests, that's probably a nightmare to maintain, because we have to duplicate some information...

Wouldn't you agree that instead of this, it would start to make sense to have a dedicated `Route` class? This class could then handle the available attributes and the regexp, and we could reuse it also in the tests. There could be a static `fromRouteConfig` factory method on the route, which could also be reused in all the tests.

And sorry again, I know that implementing it like this was my idea :grimacing: